dev/promote_bom.py

```diff
@@ -83,6 +135,7 @@ def main(cls):
     bom_promoter = cls(options)
```

Comment:
I'm not sure what promote means or how the bom gets published. It looks like promote_bom publishes it with halyard, but it doesnt specify where, so I guess halyard is configured with somewhere and this script is hoping halyard was configured correctly rather than explicitly pushing the location through.

Error detection needs to be considered in these scripts. Error handling isnt as important, though when an error is detected it should be clear there was an error and at least which component it was in and what "phase" or step you were performing so that it can be troubleshooted, replicated, fixed, and verified. You dont want to have a bunch of errors and end up completing the script and publishing a partial release.